common-core/src/main/java/com/amplifyframework/notifications/pushnotifications/NotificationPayload.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.notifications.pushnotifications
+
+import android.content.Intent
+import android.os.Parcelable
+import kotlinx.parcelize.IgnoredOnParcel
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+open class NotificationContentProvider : Parcelable {
+    @Parcelize
+    data class FCM(val content: Map<String, String>) : NotificationContentProvider()
+}
+
+@Parcelize
+open class NotificationPayload(
+    val contentProvider: NotificationContentProvider?,
+    val channelId: String? = null,
+    val targetClass: Class<*>? = null
+) : Parcelable {
+
+    @IgnoredOnParcel
+    val rawData: Map<String, String> = extractRawData()
+
+    internal constructor(builder: Builder) : this(builder.contentProvider, builder.channelId, builder.targetClass)
+
+    private fun extractRawData() = when (contentProvider) {
+        is NotificationContentProvider.FCM -> contentProvider.content
+        else -> mapOf()
+    }
+
+    companion object {
+        @JvmStatic
+        fun builder() = Builder()
```

Comment:
Do we have any other Builders where a non null type (such as contentProvider in this case) exist in the Builder constructor params, rather than an optional add on builder method?

Reply:
No. All our existing builders does not have a constructor param and all builder values are null by default.

Reply:
I think most of our other builders are really all optional items. I can't immediately think of a builder that has a required parameter. I'd like to consider adding contentProvider in the builder param.